lib/primitives/mtx.js

```diff
@@ -1881,6 +1890,132 @@ class CoinSelector {
     this.fee = Math.min(this.hardFee, CoinSelector.MAX_FEE);
     this.fund();
   }
+
+  /**
+   * Initiate selection using new selection algorithm.
+   */
+
+  async selectByValue() {
+    // add dummy output for change.
+    const change = new Output();
+
+    if (this.changeAddress) {
+      change.script.fromAddress(this.changeAddress);
+    } else {
+      // In case we don't have a change address,
+      // we use a fake p2pkh output to gauge size.
+      change.script.fromPubkeyhash(Buffer.allocUnsafe(20));
+    }
+
+    this.tx.outputs.push(change);
+
+    let target = this.outputValue;
+
+    // calculate fee for everything except inputs.
+    let size = await this.tx.estimateSize();
+    const fee = this.getFee(size);
+    // add fee to target value.
+    target += fee;
+
+    // begin selection.
+
+    // while target is greater than
+    // the largest coin we have, we
+    // will keep selecting the largest coin
+
+    this.index = this.coins.length - 1;
+    let coin = this.coins[this.index];
+    let value = Amount.value(coin.value);
+    let spendingSize = await coin.estimateSpendingSize(this.getAccount);
```

Comment:
what if `this.getAccount` is null? It's not required by coin selector (should it be?) and even `coin.estimateSpendingSize()` has a check before using it.